tests/4.5/target_update/test_target_update_to.c

```diff
@@ -1,78 +1,80 @@
-//===---- test_target_update_to.c - check the to data motion clause of target update -===//
-// 
+//===--- test_target_update_to.c ------------------------------------------===//
+//
 // OpenMP API Version 4.5 Nov 2015
-// 
+//
+// This test checks target update with the 'to' clause by checking that
+// after a target update to is applied to an array, the device reads and
+// uses the expected, new value rather than the previous value.
+//
 //===----------------------------------------------------------------------===//
 
 #include <omp.h>
 #include <stdio.h>
+#include "ompvv.h"
+
+#define N 1024
 
-#define N 100
 int a[N];
 int b[N];
 int c[N];
 
-void update_b(){
-    int i;
-    for (i = 0; i < N; i++) {
-      b[i] = b[i] * 2; 
-    }
+void update_b() {
+  int i;
+  for (i = 0; i < N; i++) {
+    b[i] = b[i] * 2;
+  }
 }
-  
+
 // Test for OpenMP 4.5 target update with to
 int main() {
-  int errors= 0, i = 0, isHost = -1, isOffloading = 0, change_flag=0;
+  int errors= 0;
+  int i = 0;
+  int isHost = -1;
+  int isOffloading = 0;
+  int change_flag = 0;
+
+  OMPVV_TEST_AND_SET_OFFLOADING(isOffloading);
 
   for (i = 0; i < N; i++) {
     a[i] = 10;
   }
 
-  // We test for offloading
-#pragma omp target map(from: isOffloading)
-  {
-    isOffloading = !omp_is_initial_device();
-  }
-
   // a and b array initialization
   for (i = 0; i < N; i++) {
-    b[i] = 2; 
+    b[i] = 2;
     c[i] = 0;
   }
+
 #pragma omp target data map(to: a[:N], b[:N]) map(from: c) map(tofrom: isHost)
-{
-  #pragma omp target 
   {
-        isHost = omp_is_initial_device();
-        int j = 0;
-        for (j = 0; j < N; j++) {
-          c[j] = (a[j] + b[j]);//c=12 
-        }
-  } // end target
-
-  update_b();
-  #pragma omp target update to(b[:N]) //update b=4 
+#pragma omp target
+    {
+      isHost = omp_is_initial_device();
+      int j = 0;
+      for (j = 0; j < N; j++) {
+        c[j] = (a[j] + b[j]);        // c = 12
+      }
+    }
 
-  #pragma omp target 
-  {
-        int j = 0;
-        for (j = 0; j < N; j++) {
-          c[j] = (c[j] + b[j]);// c=16 
-        }
-  } // end target
+    update_b();
 
-}// end target-data
+#pragma omp target update to(b[:N])  // update b = 4
 
-    // checking results 
-    for (i = 0; i < N; i++) {
-        if (c[i] != 16) {
-          errors += 1;
-        }
+#pragma omp target
+    {
+      int j = 0;
+      for (j = 0; j < N; j++) {
+        c[j] = (c[j] + b[j]);        // c = 16
+      }
     }
+  }
+
+  for (i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, c[i] != 16);
+  }
 
-  if (!errors)
-    printf("Test passed on %s.\n", (isOffloading ? "device" : "host"));
-  else
-    printf("Test failed on %s\n", (isOffloading ? "device" : "host"));
+  OMPVV_INFOMSG_IF(!errors, "Test passed on %s.\n", (isOffloading ? "device" : "host"));
+  OMPVV_ERROR_IF(errors, "Test failed on %s\n", (isOffloading ? "device" : "host"));
```

Comment:
The host/device disctintion is done by the report, based on the OMPVV_TEST_OFFLOADING result. 

These messages are not needed.